server/sonar-server/src/main/java/org/sonar/server/computation/period/PeriodsHolderImpl.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * SonarQube, open source software quality management tool.
+ * Copyright (C) 2008-2014 SonarSource
+ * mailto:contact AT sonarsource DOT com
+ *
+ * SonarQube is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * SonarQube is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package org.sonar.server.computation.period;
+
+import com.google.common.base.Preconditions;
+import java.util.ArrayList;
+import java.util.List;
+
+public class PeriodsHolderImpl implements PeriodsHolder {
+
+  private boolean isPeriodsInitialized = false;
+  private List<Period> periods = new ArrayList<>();
+
+  public void setPeriods(List<Period> periods) {
```

Comment:
there really should be some sanity check on incoming parameter

in addition, some behavior are undefined : 
* can it be set to null?
* can it be set more than once?

(I believe the answer should be NO for both questions)